### DISCUSSION TITLE
"O item já foi adicionado ao pedido" - JavaScript

### DISCUSSION THREAD

**Discussion by gjdotto** [frontendbr/forum#1734] [Perguntas e Respostas]
Olá! Sou novo nesta área e preciso de uma ajuda para realizar a seguinte funcionalidade:

No sistema da empresa, durante a realização de pedidos, é comum de existir uma duplicidade de itens adicionados. Eu preciso de um código de JavaScript para percorrer o pedido e analisar os itens e ao adicionar um item, verificar se ele já não está adicionado no pedido. Com uma mensagem por exemplo "Este item já foi adicionado, favor verificar".

Desculpem se essa pergunta já foi realizada, mas procurei diversas vezes e não consegui encontrar uma solução. Grato.

**Comment by gjdotto** [comment]
![img](https://user-images.githubusercontent.com/62561144/77344187-8f0bac80-6d11-11ea-9562-9d16741f6744.PNG)

**Comment by brunoappneta** [comment]
Olá @pdeus , em JavaScript uma forma bem genérica de checar se um item já existe em uma lista (array) é usando `find()` ou `filter()`, o primeiro pode ser usado se você precisa encontrar apenas 1 item e o segundo se você precisa encontrar/filtrar 1 ou mais itens.

Exemplo, julgando que os pedidos são um array de objetos (nesse caso, `ordersList`):

```javascript
const ordersList = [
    {
        id: 1,
        productId: 100,
        productName: 'papel higiênico'
    },
    {
        id: 2,
        productId: 200,
        productName: 'álcool em gel'
    }
]

// Procura um pedido pelo id do produto
const findOrderByProductId = productId => ordersList.find(order => order.productId === productId)

// Procura um pedido pelo nome
const findOrderByName = productName => ordersList.find(order => order.productName === productName)

console.log(findOrderByProductId(100)) // Aqui vai retornar o item 'papel higiênico'  pois ele tem o productId = 100

console.log(findOrderByProductId(500)) // Aqui vai retornar 'undefined' pois não há produto com prductId = 500 na lista

console.log(findOrderByName('álcool em gel')) // Aqui vai retornar o item 'álcool em gel' pois os nomes coincidem

console.log(findOrderByName('bolacha')) // Aqui vai retornar 'undefined' pois não existe o item 'bolacha'
```

No seu caso parece que os itens serão comparados pelo nome, então `findOrderByName` parece ser a melhor opção. Bastaria checar o retorno dentro de um `if`, por exemplo:

```javascript
const newOrderName = 'papel higiênico'
const isProductOnList = findOrderByName(newOrderName)

// Se achou um produto na lista, ele vai entrar na condição e executar o código que você precisa
if (isProductOnList) {
    alert('Este item já foi adicionado, favor verificar.')   
}
```
